src/input/hotkey.c

```diff
@@ -362,6 +381,41 @@ void hotkey_enter(void)
     }
 }
 
+void hotkey_new_map(void)
+{
+    if(editor_is_active()) {
+        // TODO: here we need the size selection for triggering window_select_list_show?
+    } else {
+        building_construction_clear_type();
+        game_undo_disable();
+        game_state_reset_overlay();
+        window_main_menu_show(1);
+    }
+}
+
+void hotkey_load(void)
+{
+    if(editor_is_active()) {
+        window_editor_map_show();
+        window_file_dialog_show(FILE_TYPE_SCENARIO, FILE_DIALOG_LOAD);
+    } else {
+        building_construction_clear_type();
+        window_city_show();
+        window_file_dialog_show(FILE_TYPE_SAVED_GAME, FILE_DIALOG_LOAD);
+    }
+}
+
+void hotkey_save(void)
+{
+    if(editor_is_active()) {
```

Comment:
Using just `editor_is_active()` will also show the dialog and return to the editor map when you for example price changes in the scenario settings open. I think it's best to check for the editor map window here.
Also, please add a space after the `if`.